lib/std/compress/deflate.zig

```diff
@@ -406,8 +417,24 @@ pub fn InflateStream(comptime ReaderType: type) type {
             if (lengths[256] == 0)
                 return error.MissingEOBCode;
 
-            try self.huffman_tables[0].construct(lengths[0..nlen]);
-            try self.huffman_tables[1].construct(lengths[nlen .. nlen + ndist]);
+            self.huffman_tables[0].construct(lengths[0..nlen]) catch |err| switch (err) {
+                error.Oversubscribed => return error.InvalidTree,
+                error.IncompleteSet => {
+                    // incomplete code ok only for single length 1 code
+                    if (nlen != self.huffman_tables[0].count[0] + self.huffman_tables[0].count[1]) {
```

Comment:
When generating test cases through fuzzing, I'm seemingly not able to find an input for which this conditional is false (and same thing for the similar conditional for distances below).

If these are not actually possible, then these `construct` calls could be simplified into simple `catch return error.InvalidTree`'s.

Reply:
That's odd, I made sure the "empty distance alphabet" test had complete code length, literal/length, and distance alphabets and tested it against puff.c, which accepts it without error. I'll have look at this more.

Reply:
Your test is totally correct, I think my wording is just confusing.

Reply:
Oh, ok, that's good. If you do still need test cases for any of the incomplete alphabet checks, they're easy to construct by hand. Let me know.